tools/ManifestSignerClient/ManifestSignerClient/Program.cs

```diff
@@ -0,0 +1,269 @@
+        {
+            if (DsaAlgorithm.Length < 2)
+            {
+                throw new Exception("DSA algorithm is not specific correctly");
+            }
+
+            string dsaAlgorithmScheme = DsaAlgorithm[0..2];
+            if (dsaAlgorithmScheme == "ES")
+            {
+                return "ES";
+            }
+            else if (dsaAlgorithmScheme == "RS")
+            {
+                return "RS";
+            }
+            else
+            {
+                throw new Exception("DSA Algorithm Type not supported");
+            }
+        }
+
+        static HashAlgorithmName GetHashAlgorithm(string DsaAlgorithm)
+        {
+            try
+            {
+                if (DsaAlgorithm.Length <= 2 || DsaAlgorithm.Length > 5)
+                {
+                    throw new Exception("SHA algorithm is not specific correctly");
+                }
+                else if (DsaAlgorithm[2..5] == "256")
+                {
+                    return HashAlgorithmName.SHA256;
+                }
+                else if (DsaAlgorithm[2..4] == "384")
+                {
+                    return HashAlgorithmName.SHA384;
+                }
+                else if (DsaAlgorithm[2..4] == "512")
+                {
+                    return HashAlgorithmName.SHA512;
+                }
+                else
+                {
+                    throw new Exception("SHA Algorithm Type not supported");
+                }
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+
+        static bool VerifyCertificate(X509Certificate2 signerCertificate, X509Certificate2 rootCertificate)
+        {
+            var ch = new X509Chain();
+            ch.ChainPolicy.ExtraStore.Add(rootCertificate);
+
+            // You can alter how the chain is built/validated. These checks are only for testing.
+            ch.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+            ch.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
+
+            // Do the validation
+            var result = ch.Build(signerCertificate);
+
+            return result;
+        }
+
+        static void Main()
+        {
+            // Check launch settings and display it
+            CheckInputAndDisplayLaunchSettings();
+
+            // Check if the given signer cert is issued by root CA
+            try
+            {
+                var signerCertificate = new X509Certificate2(SignerCertPath);
+                var rootCertificate = new X509Certificate2(DeviceRootCaPath);
+                var result = VerifyCertificate(signerCertificate, rootCertificate);
+                if (!result)
+                {
+                    throw new Exception("Please provide a signer cert issued by the given root CA");
+                }
+            }
+            catch (Exception e)
+            {
+                throw e;
```

Comment:
This would crash the console application. Instead, you should log and exit friendly. This is true for all the other exception paths as well.